images/haproxy/Dockerfile

```diff
@@ -12,7 +12,52 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# standard haproxy image + minimal config so the container will not exit
-ARG BASE="haproxy:2.2.0-alpine"
-FROM ${BASE}
+# This image is a haproxy image + minimal config so the container will not exit
+# while we rewrite the config at runtime and signal haproxy to reload.
+
+ARG BASE="k8s.gcr.io/build-image/debian-base:buster-v1.7.2"
+FROM ${BASE} as build
+
+# install:
+# - haproxy (see: https://haproxy.debian.net/)
+# - bash (ldd is a bash script and debian-base removes bash)
+# - procps (for `kill` which kind needs)
+RUN echo deb http://deb.debian.org/debian buster-backports main \
+      >/etc/apt/sources.list.d/backports.list && \
+    apt update && \
+    apt install -y --no-install-recommends haproxy=2.2.\* \
+      procps bash
+
+# copy in script for staging distro provided binary to distroless
+COPY --chmod=0755 stage-binary-and-deps.sh /usr/local/bin/
+
+# stage everything for copying into the final image
+# NOTE: kind currently also uses "mkdir" and "cp" to write files within the container
+# TODO: mkdir especially should be unnecessary, with a little refactoring
+# NOTE: kill is used to signal haproxy to reload
```

Comment:
TODO: we can probably consider dropping these in favor of docker kill/podman kill. But definitely the mkdir should be unnecessary here. I don't have a great alternative to the `cp` hack for injecting files at runtime. We could write it somewhere on the host and bind-mount but that's problematic for remote runtimes and another thing to cleanup on the host ...

Reply:
sounds good!